src/arch/linux/mapping/mapping.c

```diff
@@ -872,7 +876,7 @@ void register_hardware_ram_virtual(int type, unsigned base, unsigned int size,
 	dosaddr_t va)
 {
   register_hardware_ram_virtual2(type, base, size,
-	MEM_BASE32x(va, KVM_BASE), va);
+	MEM_BASE32x(base, KVM_BASE), va);
```

Comment:
... or possibly you can add the check
here, and fall-back to mem_base.
Then you still don't need to use _virtual2().

Reply:
And if so - fall back to mem_base.

Reply:
ok that would be this then?
`base <= ALIAS_SIZE ? MEM_BASE32x(base, KVM_BASE) : MEM_BASE32(va)`

Reply:
Maybe it should be something like
`base < mem_bases[KVM_BASE].size ? MEM_BASE32x(base, KVM_BASE) : MEM_BASE32(va)`

Reply:
ok. I think I'll just start a new PR just for mem_bases then, as it addresses another issue as well (the triple mapping).